src/components/ImageView/index.native.js

```diff
@@ -86,14 +59,56 @@ class ImageView extends PureComponent {
      * @param {GestureState} gestureState
      * @returns {Boolean}
      */
-    updatePanResponderTouches(e, gestureState) {
+    const updatePanResponderTouches = (e, gestureState) => {
         if (_.isNumber(gestureState.numberActiveTouches)) {
-            this.amountOfTouches = gestureState.numberActiveTouches;
+            numberOfTouches.current = gestureState.numberActiveTouches;
         }
 
         // We don't need to set the panResponder since all we care about is checking the gestureState, so return false
         return false;
-    }
+    };
+
+    // PanResponder used to capture how many touches are active on the attachment image
+    const panResponder = useRef(
+        PanResponder.create({
+            onStartShouldSetPanResponder: updatePanResponderTouches,
+        }),
+    ).current;
+
+    /**
+     * When the url changes and the image must load again,
+     * this resets the zoom to ensure the next image loads with the correct dimensions.
+     */
+    const resetImageZoom = useCallback(() => {
+        if (imageZoomScale.current !== 1) {
+            imageZoomScale.current = 1;
+        }
+
+        if (zoom.current) {
+            zoom.current.centerOn({
+                x: 0,
+                y: 0,
+                scale: 1,
+                duration: 0,
+            });
+        }
+    }, []);
+
+    const imageLoadingStart = () => {
+        if (isLoading) {
+            return;
+        }
+        
+        resetImageZoom();
+        setImageHeight(0);
+        setImageWidth(0);
```

Comment:
We could merge `imageHeight` and `imageWidth` into a single state, `imageDimensions`, and avoid setting state twice unnecessarily